### ISSUE TITLE
MAUI migration

### ISSUE THREAD

**Issue by VoidXH** [VoidXH/Cavern#123]
Move Cavernize GUI to a MAUI front-end to enable direct Mac/Linux builds.

**Comment by ThreeDeeJay** [comment]
Any idea if an [Android](https://learn.microsoft.com/en-us/dotnet/maui/what-is-maui) would be feasible?
![maui-overview](https://github.com/VoidXH/Cavern/assets/71472458/178fbc1b-d93a-453b-8958-0693cca84696)

**Comment by VoidXH** [comment]
It's possible, but I don't see why would you use a converter on Android.

**Comment by ThreeDeeJay** [comment]
Oh, whoops, my bad. I always get the converter mixed up with the player app.
So rephrasing as a side note, have you considered an android build for the *player*?
I know there's [Dolby Atmos for mobile devices](https://professional.dolby.com/tv/dolby-atmos-for-mobile-devices/) but only a few of devices/ROMs support/include it 😔

**Comment by VoidXH** [comment]
That's just an audio player with no video functionality, not a substitute for system-integrated Atmos virtualization.

**Comment by ThreeDeeJay** [comment]
Oh, I though the player would eventually get a video player.
but yeah, you're right. It wouldn't really be a substitute but it would be nice to have a basic decoder and virtualizer/spatializer with personalized HRTF. Hell, I would even settle for basic pHRTF virtual surround like HeSuVi, but on Android, but rn we're just stuck with Dolby Atmos or Sony 360 Reality Audio which apparently has a pHRTF generator but I'm yet to get accurate results with those compared to a few I've found going through existing ones. [Viper4Android](https://i.imgur.com/9eeebbM.mp4) and JamesDSP allow using convolvers but only for stereo sources.

**Comment by Continous** [comment]
Assuming this gets finished, would it be possible to have a Cavern Flatpak made available?

**Comment by VoidXH** [comment]
Yes, it's possible, but I really have to work out a release pipeline for that.

**Comment by VoidXH** [comment]
One problem: .NET 7 has severely reduced performance from .NET 6. .NET 8 should only be used if it results in at least equal performance.

**Comment by electrolyte-orb** [comment]
Please dont move to MAUI... its shit... i very much like the current functional design instead of a more aesthetically pleasing but complete ASS in performance

**Comment by Continous** [comment]
> Please dont move to MAUI... its shit... i very much like the old functional design instead of a more aesthetically pleasing but complete ASS in performance

Aside from "its shit" do you have any specific issues with MAUI?

**Comment by electrolyte-orb** [comment]
@Continous it's rebranded Xamarin with all the bugs...

Im not so sure about performance impact, so I don't talk about it (I'm sure you will find something interesting). But people usually choose MAUI to get lazy and write one codebase for all platforms (which is totally fine, especially when you have 2 members on your team). Well if cross platform is your concern, use https://github.com/AvaloniaUI/Avalonia

WPF is so much mature and battle tested than MAUI I will choose it everyday of the week

I just don't get why you should choose MAUI

https://github.com/dotnet/maui/discussions/10839

**Comment by VoidXH** [comment]
If it has any notable performance impact or bugs, I won't follow through with this migration, but it's worth a try to make Cavernize multiplatform.

**Comment by Continous** [comment]
Yeah, it's also worth considering that _every_ UI framework has it's shortcomings. I get not liking Maui, but being a bit more specific would really help sidestep any issues.

**Comment by electrolyte-orb** [comment]
> If it has any notable performance impact or bugs, I won't follow through with this migration, but it's worth a try to make Cavernize multiplatform.

Oh hi Sbence, first of all thank you. you are doing the lord's work here, I wish I could donate but I'm a student myself, so I'm not in position to do so.

Secondly, AvaloniaUI is much much more mature than MAUI... It's worth taking a look
https://github.com/AvaloniaUI/Avalonia

**Comment by VoidXH** [comment]
Too much effort, this time is better allocated for projects such as Filter Studio.

**Comment by herrernst** [comment]
So is compiling for Linux without GUI possible?! That would be great!

**Comment by VoidXH** [comment]
Yeah, after #235 it will be.

**Comment by herrernst** [comment]
Very nice, thank you again for your great work!
